app/routes/political/id_lookup.py

```diff
@@ -0,0 +1,163 @@
+from typing import Annotated, Any, Dict, List
+
+from fastapi import APIRouter, HTTPException, Query
+from unidecode import unidecode
+
+from app.models.pydantic.geoencoder import GeoencoderQueryParams, GeoencoderResponse
+from app.routes.datasets.queries import _query_dataset_json
+from app.settings.globals import ENV, per_env_admin_boundary_versions
+
+router = APIRouter()
+
+
+@router.get("/id-lookup", status_code=200, include_in_schema=False)
+async def id_lookup(params: Annotated[GeoencoderQueryParams, Query()]):
+    """Look up administrative boundary IDs matching a specified country name
+    (and region name and subregion name, if specified)."""
+    admin_source_to_dataset: Dict[str, str] = {"GADM": "gadm_administrative_boundaries"}
+
+    try:
+        dataset: str = admin_source_to_dataset[params.admin_source]
+    except KeyError:
+        raise HTTPException(
+            status_code=400,
+            detail=(
+                "Invalid admin boundary source. Valid sources:"
+                f" {[source for source in admin_source_to_dataset.keys()]}"
+            ),
+        )
+
+    version_str: str = lookup_admin_source_version(
+        params.admin_source, params.admin_version
+    )
+
+    names: List[str | None] = sanitize_names(
+        params.normalize_search, params.country, params.region, params.subregion
+    )
+
+    adm_level: int = determine_admin_level(*names)
+
+    sql: str = _admin_boundary_lookup_sql(
+        adm_level, params.normalize_search, dataset, *names
+    )
+
+    json_data: List[Dict[str, Any]] = await _query_dataset_json(
+        dataset, version_str, sql, None
+    )
+
+    return form_geoencoder_response(
+        params.admin_source, params.admin_version, adm_level, json_data
+    )
+
+
+def sanitize_names(
```

Comment:
What about `normalilze_names`? Sanitize comes with a different connotation about what this method does.

Reply:
Roger, will change, thanks!